pkg/providers/cloudstack/template.go

```diff
@@ -0,0 +1,70 @@
+package cloudstack
+
+import (
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/providers"
+)
+
+func generateTemplateBuilder(clusterSpec *cluster.Spec) (providers.TemplateBuilder, error) {
+	controlPlaneMachineSpec, err := getControlPlaneMachineSpec(clusterSpec)
+	if err != nil {
+		return nil, errors.Wrap(err, "generating control plane machine spec")
+	}
+
+	workerNodeGroupMachineSpecs, err := getWorkerNodeGroupMachineSpec(clusterSpec)
+	if err != nil {
+		return nil, errors.Wrap(err, "generating worker node group machine specs")
+	}
+
+	etcdMachineSpec, err := getEtcdMachineSpec(clusterSpec)
+	if err != nil {
+		return nil, errors.Wrap(err, "generating etcd machine spec")
+	}
+
+	templateBuilder := NewCloudStackTemplateBuilder(
+		&clusterSpec.CloudStackDatacenter.Spec,
+		controlPlaneMachineSpec,
+		etcdMachineSpec,
+		workerNodeGroupMachineSpecs,
+		time.Now,
+	)
```

Comment:
I think we may need to update the template builder and the way it works. It seems as though it was made prohibiting  control plane and etcd updates, and these provided arguments help to override what's in the provided spec